pkgs/servers/nextcloud/packages/apps/recognize.nix

```diff
@@ -80,25 +82,25 @@ stdenv.mkDerivation rec {
     # Replace all occurences of node (and check that we actually remved them all)
     test "$(grep "get[a-zA-Z]*('node_binary'" recognize/lib/**/*.php | wc -l)" -gt 0
     substituteInPlace recognize/lib/**/*.php \
-      --replace-quiet "\$this->settingsService->getSetting('node_binary')" "'${nodejs}/bin/node'" \
-      --replace-quiet "\$this->config->getAppValueString('node_binary', '""')" "'${nodejs}/bin/node'" \
-      --replace-quiet "\$this->config->getAppValueString('node_binary')" "'${nodejs}/bin/node'" \
-      --replace-quiet "\$this->config->getAppValue('node_binary', '""')" "'${nodejs}/bin/node'" \
-      --replace-quiet "\$this->config->getAppValue('node_binary')" "'${nodejs}/bin/node'"
+      --replace-quiet "\$this->settingsService->getSetting('node_binary')" "'${lib.getExe nodejs}'" \
+      --replace-quiet "\$this->config->getAppValueString('node_binary', '""')" "'${lib.getExe nodejs}'" \
+      --replace-quiet "\$this->config->getAppValueString('node_binary')" "'${lib.getExe nodejs}'" \
+      --replace-quiet "\$this->config->getAppValue('node_binary', '""')" "'${lib.getExe nodejs}'" \
+      --replace-quiet "\$this->config->getAppValue('node_binary')" "'${lib.getExe nodejs}'"
     test "$(grep "get[a-zA-Z]*('node_binary'" recognize/lib/**/*.php | wc -l)" -eq 0
 
 
 
     # Skip trying to install it... (less warnings in the log)
     sed  -i '/public function run/areturn ; //skip' recognize/lib/Migration/InstallDeps.php
 
-    ln -s ${ffmpeg}/bin/ffmpeg recognize/node_modules/ffmpeg-static/ffmpeg
+    ln -s ${lib.getExe ffmpeg} recognize/node_modules/ffmpeg-static/ffmpeg
   '';
 
   nativeBuildInputs = lib.optionals useLibTensorflow [
     nodejs
-    nodejs.pkgs.node-pre-gyp
-    nodejs.pkgs.node-gyp
+    node-pre-gyp
+    node-gyp
```

Comment:
I wanted the node-gyp version to correspond to the nodejs version, such that one could simply override nodejs to get the desired output result.

Reply:
Done

Reply:
So apparently this is a deprecated alias which makes evaluation fail with `allowAliases = false;` which our CI does. I have yet to check why this wasn't caught, but this is still an issue. Please change accordingly or convince the nodejs people to un-alias it.

Reply:
Reverted, should be good now?

Reply:
Yes, I managed to reproduce the earlier problem by setting `nixpkgs.config.allowAliases = false;`, and your edits indeed fix this